components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubEventPublisherImpl.java

```diff
@@ -186,22 +142,15 @@ private void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String o
         WebSubHubCorrelationLogUtils.triggerCorrelationLogForRequest(httpPost);
         final long requestStartTime = System.currentTimeMillis();
 
-        CompletableFuture<HttpResponse> future = clientManager.executeAsync(httpPost);
-
-        future.thenAccept(response -> {
-            try {
-                handleTopicMgtResponse((CloseableHttpResponse) response, httpPost, topic, operation, requestStartTime);
-            } catch (IOException | WebSubAdapterException e) {
-                log.error("Handling topic management response failed. ", e);
-            }
-        }).exceptionally(ex -> {
-            log.error("Topic management API call failed. ", ex);
-            return null;
-        });
+        try (CloseableHttpResponse response = (CloseableHttpResponse) clientManager.execute(httpPost)) {
+            handleTopicMgtResponse(response, httpPost, topic, operation, requestStartTime);
+        } catch (IOException | WebSubAdapterException e) {
+            log.error("Topic management API call failed. ", e);
+        }
```

Comment:
Shouldn't this throw an error and the flow should break right ?

Reply:
Fixed